js_modules/dagster-ui/packages/ui-core/src/util/lazy.tsx

```diff
@@ -0,0 +1,38 @@
+import React, {useEffect, useMemo, useState} from 'react';
+
+import {CompletionType, useTraceDependency} from '../performance/TraceContext';
+
+type ResolvedType<T> = T extends Promise<infer U> ? U : never;
+
+export const lazy = <T extends () => Promise<{default: React.ComponentType<any>}>>(importFn: T) => {
```

Comment:
I like this implementation and it's nice not needing a `<Suspense />` boundary around the lazy imports, but it seems like if we call it `lazy` it should match the usage of the React version.  Maybe we add a comment here explaining how this version is different from the standard one, and also forbid the import of the React export?